mesonbuild/dependencies/pkgconfig.py

```diff
@@ -32,64 +32,144 @@
     from .._typing import ImmutableListProtocol
     from ..build import EnvironmentVariables
 
-class PkgConfigDependency(ExternalDependency):
+class PkgConfigInterface:
+    '''Base class wrapping a pkg-config implementation'''
+
+    @staticmethod
+    def instance(silent: bool, env: Environment, for_machine: MachineChoice) -> T.Optional[PkgConfigInterface]:
```

Comment:
I think `silent` should be the last argument. That way, you could explicitly write `PkgConfigInterface.instance(enf, for_marchine, silent=False)`, which is self-documenting. Otherwise, we don't know what the bool flag is when the function is invoked.

The same apply for similar functions and constructors (`PkgConfigCLI`, etc.)